blueprints/02-at-scale/README.md

```diff
@@ -144,30 +155,79 @@ Once the resources have been created, a `kubeconfig` file is created in the [/k8
 
    ```sh
    eval $(terraform output --raw cbci_controllers_pods)
    ```
 
    If successful, it should indicate that 2 replicas are running for `team-c-ha` since [CloudBees CI HA/HS](https://docs.cloudbees.com/docs/cloudbees-ci/latest/ha-install-guide/) is enabled on this controller.
 
-2. Issue the following command to verify that horizontal pod autoscaling is enabled for `team-c-ha`:
+4. Issue the following command to verify that horizontal pod autoscaling is enabled for `team-c-ha`:
 
    ```sh
    eval $(terraform output --raw cbci_controller_c_hpa)
    ```
 
-3. Issue the following command to retrieve an [API token](https://docs.cloudbees.com/docs/cloudbees-ci-api/latest/api-authentication) for the `admin_cbci_a` user with the correct permissions for the required actions:
+#### Secrets Management
+
+##### Kubernetes Secret
+
+This blueprint uses a couple of Kubernetes secrets for different purposes.
+
+> [!NOTE]
+> - Beyond the CloudBees CI Addon (for demo purposes), Kubernetes secrets can be managed via [External Secret Operators](https://aws-ia.github.io/terraform-aws-eks-blueprints-addons/main/addons/external-secrets/).
+> - Kubernetes secrets could be also be retrived as Jenkins Credentials via using the [Kubernetes Credentials Provider plugin](https://jenkinsci.github.io/kubernetes-credentials-provider-plugin/).
+
+###### Casc Secrets
+
+The secrets key/value file defined in [k8s/secrets-values.yml](k8s/secrets-values.yml) is converted into a Kubernetes secret (`cbci-sec-casc`) and mounted into /run/secrets/ for Operation Center and Controllers to be consumed via CloudBees Casc. See [Configuration as Code - Handling Secrets - Kubernetes Secrets](https://github.com/jenkinsci/configuration-as-code-plugin/blob/master/docs/features/secrets.adoc#kubernetes-secrets) for more information.
+
+###### Container Registry Secrets
+
+DockerHub authentication is stored as Kubernetes secrets (`cbci-agent-sec-reg`) and mounted to [Kaniko agent containers](https://docs.cloudbees.com/docs/cloudbees-ci/latest/cloud-admin-guide/using-kaniko) to build and push images to this registry. The secret is created using the `dh_reg_secret_auth` variable.
+
+```sh
+   eval $(terraform output --raw cbci_agent_sec_reg)
+```
+
+> [!NOTE]
+> ECR authentication is done via instance profile connected to `build-linux-spot` Node pools.
+
+##### HashiCorp Vault
+
+HashiCorp Vault is used as a credential provider for CloudBees CI Pipelines in this blueprint.
+
+1. Run the configure Hashicorp Vault script. Keep in a safe place Admin Token and Unseal Keys (saved in `k8s/vault-init.log`) as well as Role ID and Secret ID for `cbci-oc` App Role.
+
+   ```sh
+   eval $(terraform output --raw vault_configure)
+   ```
+
+2. Access the HashiCorp Vault UI by issuing the following command. Enter the root token to log in from the _step 1_.
```

Comment:
We don't mention the root token in step 1 directly above. Was that something that was returned when they ran the configure Hashicorp Vault script?
```suggestion
2. Issue the following command to access the HashiCorp Vault UI. Enter the root token to sign in from _step 1_.
```